### PR TITLE
genericapiserver: simplfy EtcdOptions.Validate()

### PR DESCRIPTION
**Pull request by hongchaodeng**
Validate() only needs to return single error.
No need to complicate it with error slice.

### CODE REVIEW

**Comment by ncdc** [comment]
@deads2k @sttts

File: pkg/genericapiserver/options/etcd.go [hunk 1]
```diff
@@ -45,13 +45,12 @@ func NewEtcdOptions() *EtcdOptions {
 	}
 }
 
-func (s *EtcdOptions) Validate() []error {
-	allErrors := []error{}
+func (s *EtcdOptions) Validate() error {
```
Comment (deads2k) [inline pkg/genericapiserver/options/etcd.go]: API object validation return a list and I think that pattern works well for validation.  I think we should keep this as a slice.

Comment (hongchaodeng) [inline pkg/genericapiserver/options/etcd.go]: This isn't API object, right?

Comment (hongchaodeng) [inline pkg/genericapiserver/options/etcd.go]: I don't think this should keep the same as API object.
Why it needs to return a slice?

**Comment by deads2k** [comment]
> I don't think this should keep the same as API object.
> Why it needs to return a slice?

I think it better matches what we're looking for and we are moving towards config based on API objects.

**Comment by hongchaodeng** [comment]
@deads2k 
Sure. Thanks!

**Pull request closed without merging**